### PR TITLE
Allow to change the limit for the maximum size of the certificate chain.

### PR DESCRIPTION
**Pull request by normanmaurer**
Motivation:

Sometimes large certificate chains are used which might be larger then the default maximum size.

Modifications:

- Allow to configure the limit
- Add unit test

Result:

Be able to adjust limit

### CODE REVIEW

**Comment by normanmaurer** [comment]
This depends on https://github.com/netty/netty-tcnative/pull/784

**Review by chrisvest — APPROVED** [review]

**Comment by hyperxpro** [comment]
Why Long, not Int? Who will have more than 2 Billion certs in a single request?

File: handler/src/main/java/io/netty/handler/ssl/OpenSslContextOption.java [hunk 1]
```diff
@@ -68,4 +68,10 @@ private OpenSslContextOption(String name) {
      */
     public static final OpenSslContextOption<OpenSslCertificateCompressionConfig> CERTIFICATE_COMPRESSION_ALGORITHMS =
             new OpenSslContextOption<OpenSslCertificateCompressionConfig>("CERTIFICATE_COMPRESSION_ALGORITHMS");
+
+    /**
+     * Set the maximum number of bytes that is allowed during the handshake for certificate chain.
+     */
+    public static final OpenSslContextOption<Long> MAX_CERTIFICATE_LIST =
```
Comment (Scottmitch) [inline handler/src/main/java/io/netty/handler/ssl/OpenSslContextOption.java:75]: `MAX_CERTIFICATE_LIST` -> `MAX_CERTIFICATE_LIST_BYTES` to clarify it limits total bytes?

I assume JDK does not have any API to limit max size in bytes of cert chain?

**Review by Scottmitch — APPROVED** [review]

File: handler/src/main/java/io/netty/handler/ssl/OpenSslContextOption.java (hunk 1 above)
Comment (bryce-anderson) [reply handler/src/main/java/io/netty/handler/ssl/OpenSslContextOption.java:75]: +1

**Review by bryce-anderson — APPROVED** [review]

**Review by hyperxpro — APPROVED** [review]
Thanks for addressing this. :)

**Pull request merged**